### PR TITLE
feat: add private and anonymous analytics

### PR DESCRIPTION
**Pull request by janek26**

### CODE REVIEW

File: packages/extension/.env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+SEGMENT_WRITE_KEY=
```
Comment (delaaxe) [inline packages/extension/.env.example:1]: great practice

File: packages/extension/src/shared/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      }
+}
+
+export interface Pages {
+  welcome: undefined
+  disclaimer: undefined
+  createWallet: undefined
+  restoreWallet: undefined
+  restoreWalletWithFile: undefined
+}
```
Comment (delaaxe) [inline packages/extension/src/shared/analytics.ts:28]: this is a bit weird, couldn't it be expressed as a union of strings?

File: packages/extension/src/shared/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  page<T extends keyof Pages>(
+    name: T,
+    ...rest: Pages[T] extends undefined ? [data?: Pages[T]] : [data: Pages[T]]
+  ): Promise<unknown>
+}
+
+export type Fetcher = (url: string, init?: RequestInit) => Promise<unknown>
+
+export function getAnalytics(fetch: Fetcher): Analytics {
+  const defaultPayload = {
```
Comment (delaaxe) [inline packages/extension/src/shared/analytics.ts:46]: could be moved outside of function

File: packages/extension/src/shared/analytics.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      ? [data?: Events[T]]
+      : [data: Events[T]]
+  ): Promise<unknown>
+  page<T extends keyof Pages>(
+    name: T,
+    ...rest: Pages[T] extends undefined ? [data?: Pages[T]] : [data: Pages[T]]
+  ): Promise<unknown>
+}
+
+export type Fetcher = (url: string, init?: RequestInit) => Promise<unknown>
```
Comment (delaaxe) [inline packages/extension/src/shared/analytics.ts:43]:
```suggestion
export type Fetch = (url: string, init?: RequestInit) => Promise<unknown>
```

File: packages/extension/src/ui/services/analytics.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      } else {
+        return fetch(url, init)
+      }
+    } 
+  * ```
+*/
+
+export const analytics = getAnalytics(fetch)
+
+export const usePageTrack = <T extends keyof Pages>(
```
Comment (delaaxe) [inline packages/extension/src/ui/services/analytics.ts:28]:
```suggestion
export const usePageTracking = <T extends keyof Pages>(
```

**Comment by janek26** [comment]
great idea!

**Review by delaaxe — APPROVED** [review]

**Pull request merged**